### PR TITLE
'Thats The Way' exercise finished

### PR DESCRIPTION
**Pull request by michall1900**
Hey itayday,
Please check the exercise,
Thanks in advance .

### CODE REVIEW

**Comment by talbarfm** [comment]
You have conflicts ! make sure to fix them!

**Comment by talbarfm** [comment]
Pay attention to add a capital letter to the commit message

File: 5.1 Ex/5.1 Thats The Way.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import os
```
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: Add a newline the imports and the consts

File: 5.1 Ex/5.1 Thats The Way.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+FILES_PREFIX = "deep"
+
+
+def find_files_at_path_with_wanted_prefix(path='.', prefix=FILES_PREFIX):
```
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: thats a way too long function name - find_files_with_prefix should suffice

File: 5.1 Ex/5.1 Thats The Way.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    The function receives path to search in and prefix to search files with that prefix.
+    The function return list of files name's that starting with thw wanted prefix.
+    Note: I searched and find the solution at this link:
+    https://stackoverflow.com/questions/3964681/find-all-files-in-a-directory-with-extension-txt-in-python
+
+    @param path: String, for search at wanted path.
+    @param prefix: String, for search files with that prefix
+    @return: List of files name's that their prefix is as wanted that existing in wanted path.
+    """
+    files_list_with_wanted_suffix = [file for file in os.listdir(path) if file.startswith(prefix)]
```
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: just return the new list - you dont  have to assign it to a variable

File: 5.1 Ex/5.1 Thats The Way.py (hunk 2 above)
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: I would use partial for this use, or wrap the function with another function that is suited for this exercise requests - youve made a generic function but the default values are not generic therefore should not be there

File: 5.1 Ex/5.1 Thats The Way.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    return files_list_with_wanted_suffix
+
+
+wanted_path = input("Please Enter wanted path: ")
```
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: wrap all of this inside a function - you can name it main, you can name it what ever you want, make sure its not executed when this file is imported

File: 5.1 Ex/5.1 Thats The Way.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+files_list = find_files_at_path_with_wanted_prefix(path=wanted_path)
+print("The list is:")
```
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: use list comprehension instead of this for loop

File: 5.1 Ex/5.1 Thats The Way.py (hunk 2 above)
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: Add type hints instead of documenting the file type in a comment

Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: Add return type to the funciton. the -> syntax

File: 5.1 Ex/5.1 Thats The Way.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import os
+FILES_PREFIX = "deep"
+
+
+def find_files_at_path_with_wanted_prefix(path='.', prefix=FILES_PREFIX):
+    """
+    The function receives path to search in and prefix to search files with that prefix.
```
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: You can join these two sentences to one and remove the second "The funciton"

File: 5.1 Ex/5.1 Thats The Way.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import os
+FILES_PREFIX = "deep"
```
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: Why is this a const if you use it once?

**Review by talbarfm — CHANGES_REQUESTED** [review]

File: 5.1 Ex/5.1 Thats The Way.py (hunk 4 above)
Comment (michall1900) [reply 5.1 Ex/5.1 Thats The Way.py]: Tal

**Comment by talbarfm** [comment]
@itayday

File: 5.1 Ex/5.1 Thats The Way.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
+"""Searching at given path files with given prefix, create a list of files name's
```
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: Add a newline after """

File: main.py [hunk 9]
```diff
@@ -0,0 +1 @@
+
```
Comment (talbarfm) [inline main.py]: ????? what is this file doint here?

File: 5.1 Ex/5.1 Thats The Way.py [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    r"""Send wanted path and prefix to find_files_with_prefix function.
+    @return:  List of files name's with prefix deep inside "yam mesika\ ... \images" directory.
+    """
+    return find_files_with_prefix(path=r".\yam mesika\Notebooks\week05\images", prefix="deep")
+
+
+def main_that_is_the_way():
+    files_list = send_specific_data()
+    print("Files name's are:")
+    [print(file_name) for file_name in files_list]
```
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: oh no no no 
https://stackoverflow.com/questions/5753597/is-it-pythonic-to-use-list-comprehensions-for-just-side-effects

File: 5.1 Ex/5.1 Thats The Way.py [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def find_files_with_prefix(path: str, prefix: str) -> list:
+    """ Receives path to search in and prefix to search files with that prefix.
+    Return list of files name's that starting with the wanted prefix.
+    Note: I searched and find the solution at this link:
+    https://stackoverflow.com/questions/3964681/find-all-files-in-a-directory-with-extension-txt-in-python
+
+    @param path: For search at wanted path.
+    @param prefix: For search files with that prefix
```
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: . in the end

File: 5.1 Ex/5.1 Thats The Way.py (hunk 8 above)
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: You dont have to add this description,  the function description is enough

File: 5.1 Ex/5.1 Thats The Way.py [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+    @return: List of files name's that their prefix is as wanted that existing in wanted path.
+    """
+    return [file for file in listdir(path) if file.startswith(prefix)]
+
+
+def send_specific_data() -> list:
```
Comment (talbarfm) [inline 5.1 Ex/5.1 Thats The Way.py]: I would replace this function with a partial that sets the prefix to deep, anyway, good thinking! the coupling to the specific path is redundant in my opinion

**Review by talbarfm — CHANGES_REQUESTED** [review]
Good job!

**Review by lovodkin93 — APPROVED** [review]
Great job and great documentation!

**Pull request merged**